src/server/pfs/fuse/filesystem.go

```diff
@@ -289,7 +297,7 @@ func (h *handle) Write(ctx context.Context, request *fuse.WriteRequest, response
 	// observed on osx, not on linux.
 	repeated := h.written - int(request.Offset)
 	if repeated < 0 {
-		return fmt.Errorf("gap in bytes written, (OpenNonSeekable should make this impossible)")
+		repeated = h.written
```

Comment:
It happens whenever you are appending to an existing file for the first time.  You get `h.written = 0` and `request.Offset = len(file)` which results in `repeated < 0`.